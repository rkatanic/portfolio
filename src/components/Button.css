.btn {
  display: flex;
  align-items: center;
  padding: 1rem 2rem;
  border: none;
  font-size: var(--font-size-1-25);
  font-family: var(--body-font);
  font-weight: 600;
  cursor: pointer;
  transition: 0.4s;
}

.btn:hover {
  transform: translateY(-2px);
  opacity: 0.9;
}

.btn:disabled {
  cursor: not-allowed;
  background-color: var(--color-text-light);
}

.btn-primary {
  color: #fff;
  background-color: var(--color-primary);
}

.btn-primary .icon {
  fill: #fff;
  height: 1.5rem;
  margin-left: 0.5rem;
}

.btn-outline {
  border: none;
  color: var(--color-title);
  background-color: var(--color-container);
}

.btn-outline .icon {
  fill: var(--color-title);
  margin-left: 0.5rem;
}
