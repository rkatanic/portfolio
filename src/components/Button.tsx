import "./Button.css";

interface Props {
  onClick?: () => void;
  disabled?: boolean;
  type?: "button" | "submit";
  variant?: "btn-primary" | "btn-outline";
  label: string | JSX.Element;
}

const Button = ({
  onClick,
  disabled,
  type = "button",
  variant = "btn-primary",
  label,
}: Props): JSX.Element => (
  <button
    disabled={disabled}
    className={`btn ${variant}`}
    onClick={onClick}
    type={type}
  >
    <span className="txt-xs txt-600">{label}</span>
  </button>
);

export default Button;
