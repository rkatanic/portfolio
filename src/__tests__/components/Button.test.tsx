import { fireEvent, render } from "@testing-library/react";
import Button from "../../components/Button";

describe("Button", (): void => {
  it("should render primary button as default", (): void => {
    const { baseElement } = render(<Button>test button</Button>);

    expect(baseElement).toMatchSnapshot();
  });

  it("should render outline button", (): void => {
    const { baseElement } = render(
      <Button variant="btn-outline">test button</Button>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should render button type as default", (): void => {
    const { getByText } = render(<Button>test button</Button>);

    expect(getByText("test button")).toHaveAttribute("type", "button");
  });

  it("should handle onClick function", (): void => {
    const mockOnClick = jest.fn();
    const { baseElement, getByText } = render(
      <Button onClick={mockOnClick}>test button</Button>
    );

    fireEvent.click(getByText("test button"));

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
