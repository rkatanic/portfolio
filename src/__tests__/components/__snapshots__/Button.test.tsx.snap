// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render outline button 1`] = `
<body>
  <div>
    <button
      class="btn btn-outline"
      type="button"
    >
      test button
    </button>
  </div>
</body>
`;

exports[`Button should render primary button as default 1`] = `
<body>
  <div>
    <button
      class="btn btn-primary"
      type="button"
    >
      test button
    </button>
  </div>
</body>
`;
